docs/process/release-planning.md

```diff
@@ -1,7 +1,71 @@
 # Release Planning
 
-We plan work at two levels of granularity: first at the marketing release level (1.1, 1.2, etc.) and then at the milestone level. Marketing releases are made up of multiple milestones. Milestones should contain roughly a month’s worth of work.
+This document outlines our process for planning and scheduling releases, so you
+can familiarize yourself with the flow of work from opening an issue to seeing
+it published in a release.
 
-We’ll release updates as needed or when a milestone is completed.
+## Releases
 
-Large features and user-visible changes are feature-flagged, to be unflagged in the next marketing release. Beta and dev builds will have all feature flags enabled.
+We organize releases in two ways - marketing and milestones:
+
+ - Marketing releases are what we use to represent planned features, and are
+   high-level goals
+      - **e.g. 1.4, 1.5, etc**
+ - Milestone releases are used to track issues and pull requests associated
+   with a marketing release, and can be followed along on GitHub
+      - **e.g. 1.4.1, 1.4.2, 1.5.0, etc**
+
+We aim to ship updates to production every two weeks, to ensure a continuous
+flow of improvements to our users. Track our progress in the [current milestones](https://github.com/desktop/desktop/milestones?direction=desc&sort=completeness&state=open).
+
+## Scheduling Pull Requests
+
+Pull Requests for user-facing changes should have a milestone associated with
+it, to indicate when it should be merged.
+
+### Features
+
+Pull Requests associated with features for our marketing releases should have a
```

Comment:
Is there something here about the associated issue also being having the milestone defined, or is that covered by "the reviewer should also ensure any issues linked..." below? I think it might be worth spelling out here.

Reply:
I haven't thought much about what to do with issues, but basically I'd love to support and tweak our current flow:

 - PR descriptions should link to issues
 - automation to pull linked issues into the right milestone as they are assigned or unassigned by the reviewer (rather than needing someone to do it manually and potentially mess something up)

For now I'm 👍 to the manual step I guess, unless others have stronger feels about one or the other.

Reply:
I made a change in cce8e9d874b059f537edb4856ab7739b76596c8f to make this clearer. Let me know if there's anything more than this.

> The maintainer merging the pull request should also ensure any issues linked in the pull request description (which will be auto-closed when merging) are also assigned to the same milestone.